app/Http/Controllers/ContactController.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use Mail;
+
+class ContactController extends Controller
+{
+    public function contact(){
+        return view('contact');
+    }
+
+    public function processData(Request $request){
+        $this->validate($request, [
+                'fname' => 'required',
+                'lname' => 'required',
+                'email' => ['required','email'],
+                'phone' => ['required','numeric','digits_between:9,12'],
+                'message' => 'required'
+        ]);
+
+        Mail::send('emails.contact-tmpl', [
+                'name' => $request->get('fname')." ".$request->get('lname'),
+                'email' => $request->get('email'),
+                'phone' => $request->get('phone'),
+                'project' => $request->get('project'),
+                'service' => $request->get('service'),
+                'estimate' => $request->get('estimate'),
```

Comment:
what is it used for?

Reply:
Reading the form submit values from http request.